core/src/test/java/io/parsingdata/metal/data/SelectionTest.java

```diff
@@ -41,6 +41,9 @@ private Source createSource() {
         return new Source() {
             @Override protected byte[] getData(BigInteger offset, BigInteger length) { return new byte[0]; }
             @Override protected boolean isAvailable(BigInteger offset, BigInteger length) { return true; }
+            @Override public int immutableHashCode() { return 0; }
+            @Override
+            public boolean equals(Object obj) {return obj == this;}
```

Comment:
Minor style consistency nitpick:

```suggestion
            public boolean equals(Object obj) { return obj == this; }
```